BotBrownCore/Workers/TextToSpeech/TextToSpeechProcessor.cs

```diff
@@ -1,150 +1,150 @@
-        private readonly AudioConfiguration audioConfiguration;
-        private bool isInitialized;
-        private string languages;
-
-        public TextToSpeechProcessor(IConfigurationManager configurationManager)
-        {
-            this.configurationManager = configurationManager;
-            RegisterAvailableLanguages();
-
-            audioConfiguration = configurationManager.LoadConfiguration<AudioConfiguration>(ConfigurationFileConstants.Audio);
-        }
-
-        public string TextToSpeechLanguages
-        {
-            get
-            {
-                if (string.IsNullOrEmpty(languages))
-                {
-                    InitializeLanguages();
-                }
-
-                return languages;
-            }
-        }
-
-        public void Speak(ChannelUser user, Func<ChannelUser, string> messageAction)
-        {
-            bool isTextToSpeechActive = CheckIfTextToSpeechIsActive();
-            if (!isTextToSpeechActive)
-            {
-                return;
-            }
-
-            audioConfiguration.InitializeConfiguration();
-
-            using (var stream = new MemoryStream())
-            using (var output = new WasapiOut(audioConfiguration.SelectedTTSDevice, AudioClientShareMode.Shared, false, 100))
-            using (var provider = new RawSourceWaveStream(stream, new WaveFormat(44100, BitResolution, 2)))
-            using (var synth = new SpeechSynthesizer())
-            {
-                synth.Volume = 100;
-                synth.SetOutputToAudioStream(stream, new SpeechAudioFormatInfo(44100, AudioBitsPerSample.Sixteen, AudioChannel.Stereo));
-                synth.SelectVoice(GetDesiredLanguage(user));
-                synth.Speak(messageAction(user));
-
-                stream.Seek(0, SeekOrigin.Begin);
-
-                output.Volume = 1f;
-                output.Init(provider);
-                output.Play();
-
-                while (output.PlaybackState == PlaybackState.Playing)
-                {
-                    Thread.Sleep(10);
-                }
-            }
+﻿namespace BotBrown.Workers.TextToSpeech
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.IO;
+    using System.Speech.AudioFormat;
+    using System.Speech.Synthesis;
+    using System.Text;
+    using System.Threading;
+    using BotBrown.Configuration;
+    using BotBrownCore.Configuration;
+    using NAudio.CoreAudioApi;
+    using NAudio.Wave;
+
+    public class TextToSpeechProcessor : ITextToSpeechProcessor
+    {
+        private const int BitResolution = 16;
+
+        private readonly IConfigurationManager configurationManager;
+        private readonly IDictionary<string, string> availableLanguages = new Dictionary<string, string>();
+        private readonly AudioConfiguration audioConfiguration;
+        private bool isInitialized;

```

Comment:
isInitialized isn't used actually. Didn't we dropped the init process?

Reply:
Yes. I think we did. I'll fix it with the issue regarding the other warnings.